SignalServiceKit/src/Messages/OWSReadReceiptManager.m

```diff
@@ -445,6 +412,148 @@ - (void)updateOutgoingMessageFromLinkedDevice:(TSOutgoingMessage *)message
     [TSRecipientReadReceipt removeRecipientIdsForTimestamp:message.timestamp transaction:transaction];
 }
 
+#pragma mark - Linked Device Read Receipts
+
+- (void)updateIncomingMessage:(TSIncomingMessage *)message
+                  transaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    OWSAssert(message);
+    OWSAssert(transaction);
+
+    NSString *senderId = message.messageAuthorId;
+    uint64_t timestamp = message.timestamp;
+    if (senderId.length < 1 || timestamp < 1) {
+        OWSFail(@"%@ Invalid incoming message: %@ %llu", self.tag, senderId, timestamp);
+        return;
+    }
+
+    OWSLinkedDeviceReadReceipt *_Nullable readReceipt = [OWSLinkedDeviceReadReceipt linkedDeviceReadReceiptWithSenderId:senderId
```

Comment:
significant: Shouldn't this take in and use the transaction?
nit: This is a finder method but reads like a factory method

Reply:
> significant: Shouldn't this take in and use the transaction?

Oddly, the nested transaction worked fine?  Anyway, I've plumbed through the transaction.

> nit: This is a finder method but reads like a factory method

Renamed.